packages/material-ui-lab/src/useAutocomplete/useAutocomplete.d.ts

```diff
@@ -26,6 +26,10 @@ export type Value<T, Multiple, DisableClearable, FreeSolo> = Multiple extends un
     : T | null | AutocompleteFreeSoloValueMapping<FreeSolo>
   : Array<T | AutocompleteFreeSoloValueMapping<FreeSolo>>;
 
+export interface Option {
```

Comment:
I think that we should aim for making type names unique
```suggestion
export interface AutocompleteOption {
```

Reply:
This would allow the following code that is not sound:
```tsx
const firstItem = { title: "second" };
const secondItem: Option = { label: "first" };
<AutoComplete
  options={[firstItem, secondItem]}
  onChange={(event, option) => {
    // duh!
    console.log(option.title);
  }}
/>;
```
does code completion work if we default `T = AutoCompleteOption`?

Reply:
Replacing T with AutoCompleteOption will work but it is a breaking change and having an array of strings `['one', 'two']` will stop working.

Reply:
Was meant to comment on https://github.com/mui-org/material-ui/pull/21992#discussion_r463167041.